beacon-chain/db/kv/state_summary.go

```diff
@@ -78,9 +82,6 @@ func (s *Store) HasStateSummary(ctx context.Context, blockRoot [32]byte) bool {
 }
 
 func (s *Store) hasStateSummaryBytes(tx *bolt.Tx, blockRoot [32]byte) bool {
-	if s.stateSummaryCache.has(blockRoot) {
```

Comment:
could we actually keep the check in here, but instead of passing in a tx, this method creates its own tx (in a View - so it would be a read tx)?

Reply:
I mainly suggest this because I think keeping the cache with the db read is better for consistency, less likely for someone to forget to check the cache in new code going forward. I'm not sure if the possible additional read-only tx is more overhead, but I suspect its better to move this work into a read-only tx (which doesn't block writers) vs doing extra work inside a read-write tx.

Reply:
Removed this method entirely and used HasStateSummary